### ISSUE TITLE
CI failure on GHC 8.6.5 only (`libz.so: cannot open shared object file: No such file or directory`)

### ISSUE THREAD

**Issue by RyanGlScott** [GaloisInc/what4#169]
The GHC 8.6.5 job (and _only_ the 8.6.5 job) is failing on `master` currently due to `zlib`-related issues ([example](https://github.com/GaloisInc/what4/runs/4293177190?check_suite_focus=true#step:12:944)):

```
[ 2 of 76] Compiling What4.BaseTypes  ( src/What4/BaseTypes.hs, /home/runner/work/what4/what4/dist-newstyle/build/x86_64-linux/ghc-8.6.5/what4-1.2.1.0.99/build/What4/BaseTypes.o )
<command line>: can't load .so/.DLL for: libz.so (libz.so: cannot open shared object file: No such file or directory)
```

**Comment by RyanGlScott** [comment]
I suspected that this might due to a stale build cache, but even after bumping the cache number in #170, the problem still persists.

**Comment by RyanGlScott** [comment]
Actually, we are already doing just that [here](https://github.com/GaloisInc/what4/blob/dd0755f083e9a611850a5433cfbdd520fd1547ec/.github/workflows/test.yml#L108). Hm.  In that case, I'm stumped. Any idea, @kquick?

**Comment by RyanGlScott** [comment]
Fixed in #171.
